@nrwl/cypress not installing cypress

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
<!-- What is the behavior that currently you experience? -->

When adding `@nrwl/cypress` and generating a new e2e project, it does not add cypress itself to the dependencies.

## Expected Behavior
<!-- What is the behavior that you expect to happen? -->
<!-- Is this a regression? .i.e Did this used to be the behavior at one point?  -->

When using `@nrwl/cypress`, running e2e should work OOTB without installing cypress manually, at least without concrete warnings in the CLI that it's supposed to be a manual step.

## Steps to Reproduce
<!-- Help us help you by making it easy for us to reproduce your issue! -->

```sh
npx create-nx-workspace@14.8.4 test-repo # version is 14.8.4
cd test-repo
npm add -D @nrwl/cypress
npx nx generate @nrwl/cypress:cypress-project test-1 --baseUrl=localhost:3000
```

<!-- Can you reproduce this on https://github.com/nrwl/nx-examples? -->
<!-- If so, open a PR with your changes and link it below. -->
<!-- If not, please provide a minimal Github repo -->
<!-- At the very least, provide as much detail as possible to help us reproduce the issue -->

### Failure Logs
<!-- Please include any relevant log snippets or files here. -->

```
> nx run test-1:e2e

 [31m>[39m [0m[7m[1m[31m NX [39m[22m[27m[0m  [31mCannot find module 'cypress'[39m
   Require stack:
   - /Users/gioraguttsait/Git/POCs/shopify-cypress-poc/node_modules/@nrwl/cypress/src/executors/cypress/cypress.impl.js
   - /Users/gioraguttsait/Git/POCs/shopify-cypress-poc/node_modules/nx/src/config/workspaces.js
   - /Users/gioraguttsait/Git/POCs/shopify-cypress-poc/node_modules/nx/src/command-line/run.js
   - /Users/gioraguttsait/Git/POCs/shopify-cypress-poc/node_modules/nx/bin/run-executor.js
   Pass --verbose to see the stacktrace.

 

 >  NX   Running target "test-1:e2e" failed

   Failed tasks:
   
   - test-1:e2e
   
   Hint: run the command with --verbose for more details.
```

### Environment
<!-- It's important for us to know the context in which you experience this behavior! -->
<!-- Please paste the result of `nx report` below! -->

```
 >  NX   Report complete - copy this into the issue template

   Node : 14.19.3
   OS   : darwin x64
   npm  : 6.14.17
   
   nx : 14.8.4
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.8.4
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.8.4
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 14.8.4
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 14.8.4
   @nrwl/js : Not Found
   @nrwl/linter : 14.8.4
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : Not Found
   @nrwl/workspace : 14.8.4
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

--- a/packages/cypress/src/generators/cypress-project/cypress-project.spec.ts
+++ b/packages/cypress/src/generators/cypress-project/cypress-project.spec.ts
@@ -11,9 +11,10 @@ import { cypressProjectGenerator } from './cypress-project';
 import { Schema } from './schema';
 import { Linter } from '@nrwl/linter';
 import { installedCypressVersion } from '../../utils/cypress-version';
+import { cypressInitGenerator } from '../init/init';
 
 jest.mock('../../utils/cypress-version');
-
+jest.mock('../init/init');
 describe('Cypress Project', () => {
   let tree: Tree;
   const defaultOptions: Omit<Schema, 'name' | 'project'> = {
@@ -23,6 +24,8 @@ describe('Cypress Project', () => {
   let mockedInstalledCypressVersion: jest.Mock<
     ReturnType<typeof installedCypressVersion>
   > = installedCypressVersion as never;
+  let mockInitCypress: jest.Mock<ReturnType<typeof cypressInitGenerator>> =
+    cypressInitGenerator as never;
 
   beforeEach(() => {
     tree = createTreeWithEmptyV1Workspace();
@@ -54,6 +57,26 @@ describe('Cypress Project', () => {
     });
   });
   afterEach(() => jest.clearAllMocks());
+
+  it('should call init if cypress is not installed', async () => {
+    mockedInstalledCypressVersion.mockReturnValue(null);
+    await cypressProjectGenerator(tree, {
+      ...defaultOptions,
+      name: 'my-app-e2e',
+      project: 'my-app',
+    });
+    expect(mockInitCypress).toHaveBeenCalled();
+  });
+
+  it('should call not init if cypress is installed', async () => {
+    mockedInstalledCypressVersion.mockReturnValue(10);
+    await cypressProjectGenerator(tree, {
+      ...defaultOptions,
+      name: 'my-app-e2e',
+      project: 'my-app',
+    });
+    expect(mockInitCypress).not.toHaveBeenCalled();
+  });
 
   describe('> v10', () => {
     beforeEach(() => {

--- a/packages/cypress/src/generators/cypress-project/cypress-project.ts
+++ b/packages/cypress/src/generators/cypress-project/cypress-project.ts
@@ -27,6 +27,7 @@ import {
   cypressVersion,
   eslintPluginCypressVersion,
 } from '../../utils/versions';
+import { cypressInitGenerator } from '../init/init';
 // app
 import { Schema } from './schema';
 
@@ -248,13 +249,21 @@ export async function addLinter(host: Tree, options: CypressProjectSchema) {
 
 export async function cypressProjectGenerator(host: Tree, schema: Schema) {
   const options = normalizeOptions(host, schema);
+  const tasks = [];
+  const cypressVersion = installedCypressVersion();
+  // if there is an installed cypress version, then we don't call
+  // init since we want to keep the existing version that is installed
+  if (!cypressVersion) {
+    tasks.push(cypressInitGenerator(host, options));
+  }
   createFiles(host, options);
   addProject(host, options);
   const installTask = await addLinter(host, options);
+  tasks.push(installTask);
   if (!options.skipFormat) {
     await formatFiles(host);
   }
-  return installTask;
+  return runTasksInSerial(...tasks);
 }
 
 function normalizeOptions(host: Tree, options: Schema): CypressProjectSchema {

